src/android/FirebaseMessagingPlugin.java

```diff
@@ -137,7 +142,14 @@ private void requestPermission(CordovaArgs args, CallbackContext callbackContext
         if (NotificationManagerCompat.from(context).areNotificationsEnabled()) {
             callbackContext.success();
         } else {
-            callbackContext.error("Notifications permission is not granted");
+            if (Build.VERSION.SDK_INT >= 32) {
+                if (ContextCompat.checkSelfPermission(null,
+                        Manifest.permission.ACCESS_NOTIFICATION_POLICY) == PackageManager.PERMISSION_GRANTED)
+                    return;
+                firebaseMessagingActivity.launchPermissions();
```

Comment:
Use private permission launcher and store `calbackContext ` for later usage
```suggestion
                requiestPermissionCallback = calbackContext;
                requestPermissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS);
```